### ISSUE TITLE
connection refused when updating index

### ISSUE THREAD

**Issue by Sw3d15h-F1s4** [arduino/arduino-cli#2158]
### Describe the problem

When running `arduino-cli core update-index`

`Downloading index: library_index.tar.bz2 Get "https://downloads.arduino.cc/libraries/library_index.tar.bz2": dial tcp: lookup downloads.arduino.cc on [::1]:53: read udp [::1]:54136->[::1]:53: read: connection refused
Error updating indexes: Error downloading index 'https://downloads.arduino.cc/libraries/library_index.tar.bz2': Get "https://downloads.arduino.cc/libraries/library_index.tar.bz2": dial tcp: lookup downloads.arduino.cc on [::1]:53: read udp [::1]:54136->[::1]:53: read: connection refused`

### To reproduce

installed linux arm64 binary from arduino.github.io's installation instructions, moved it to /bin and executed. seems to work fine otherwise. ran arduino-cli config init as well.

### Expected behavior

i expect it to update the board index

### Arduino CLI version

arduino-cli  Version: 0.32.2 Commit: 2661f5d9 Date: 2023-04-12T09:41:53Z

### Operating system

Linux

### Operating system version

linux aarch64 5.4.147-qgki-25884040-abT733XXU2CWC5

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-cli/issues?q=)
- [X] I verified the problem still occurs when using the [nightly build](https://arduino.github.io/arduino-cli/dev/installation/#nightly-builds)
- [X] My report contains all necessary details

**Comment by cmaglie** [comment]
This looks like a network configuration problem:

```
dial tcp: lookup downloads.arduino.cc on [::1]:53: read udp [::1]:54136->[::1]:53: read: connection refused
```

Arduino CLI cannot resolve the name `downloads.arduino.cc`.
Does `ping downloads.arduino.cc` works?

**Comment by codeandfire** [comment]
Hi, I'm having the same issue; tried the Nightly version, v0.32.2, and v0.31.0.

`ping downloads.arduino.cc` works, though it seems to be a bit slow:
```
PING downloads.arduino.cc.cdn.cloudflare.net (104.18.12.241) 56(84) bytes of data.
64 bytes from 104.18.12.241 (104.18.12.241): icmp_seq=1 ttl=53 time=38.7 ms
64 bytes from 104.18.12.241 (104.18.12.241): icmp_seq=2 ttl=53 time=41.3 ms
64 bytes from 104.18.12.241 (104.18.12.241): icmp_seq=3 ttl=53 time=39.7 ms
^C64 bytes from 104.18.12.241: icmp_seq=4 ttl=53 time=37.8 ms

--- downloads.arduino.cc.cdn.cloudflare.net ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 23270ms
rtt min/avg/max/mdev = 37.827/39.388/41.293/1.288 ms
```

I'm also able to manually download that file (`https://downloads.arduino.cc/libraries/library_index.tar.bz2`), if that helps.

**Comment by cmaglie** [comment]
Do you have an error similar to the following?

```
dial tcp: lookup downloads.arduino.cc on [::1]:53: read udp [::1]:54136->[::1]:53: read: connection refused
```

if yes, could you paste the output of:

```
sudo netstat -puntl
```

and

```
dig downloads.arduino.cc
```

**Comment by codeandfire** [comment]
@cmaglie thanks for responding.

> Do you have an error similar to the following?
> 
> ```
> dial tcp: lookup downloads.arduino.cc on [::1]:53: read udp [::1]:54136->[::1]:53: read: connection refused
> ```

Yes, precisely.

> if yes, could you paste the output of:
> 
> ```
> sudo netstat -puntl
> ```

```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.54:53           0.0.0.0:*               LISTEN      428/systemd-resolve 
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      428/systemd-resolve 
tcp        0      0 0.0.0.0:5355            0.0.0.0:*               LISTEN      428/systemd-resolve 
tcp6       0      0 :::5355                 :::*                    LISTEN      428/systemd-resolve 
udp        0      0 127.0.0.54:53           0.0.0.0:*                           428/systemd-resolve 
udp        0      0 127.0.0.53:53           0.0.0.0:*                           428/systemd-resolve 
udp        0      0 192.168.1.7:68          0.0.0.0:*                           488/iwd             
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           428/systemd-resolve 
udp        0      0 0.0.0.0:5355            0.0.0.0:*                           428/systemd-resolve 
udp6       0      0 :::5353                 :::*                                428/systemd-resolve 
udp6       0      0 :::5355                 :::*                                428/systemd-resolve 
```

> and
> 
> ```
> dig downloads.arduino.cc
> ```

```
;; communications error to ::1#53: connection refused
;; communications error to ::1#53: connection refused
;; communications error to ::1#53: connection refused
;; communications error to 127.0.0.1#53: connection refused

; <<>> DiG 9.18.14 <<>> downloads.arduino.cc
;; global options: +cmd
;; no servers could be reached
```

**Comment by cmaglie** [comment]
Also knowing the Linux distribution may help to find if there are open bugs.

**Comment by codeandfire** [comment]
Thank you for the explanation! I have a fresh Arch Linux installation so `/etc/resolv.conf` is basically empty:

```
$ cat /etc/resolv.conf 
# Resolver configuration file.
# See resolv.conf(5) for details.
```

Is that the problem? I am not facing network issues anywhere else - am able to browse normally, etc. - so I wasn't suspecting network configuration issues on my end.

**Comment by Sw3d15h-F1s4** [comment]
Ok, so after some searching, reading, and learning, I think I know what my issue is. I'm trying to run the arduino cli on Android through Termux, which has many differences from regular Linux (such as no etc/resolv.conf and using different c libraries instead of gcc). Now I don't need to compile, I was just trying to use arduino-cli to set up arduino-language-server for neovim (surprisingly neovim is quite nice to use on Android, other than lsp installation and setup being harder than usual) since I wanted to be able to contribute to my team's arduino sketches while on the move at college. Yeah, ik, dumb to run this on android, whatever. For anyone in the same boat, platformio works a little better for Android since it uses regular .cpp files, and you can set up clangd or whatever else like normal (only get a mild issue of no thread api but again you shouldn't need to compile from an android device, like how are you going to push code to the board your USB ports are a challenge to access through termux). Anyway, sorry for the trouble. I'm a noob to Linux, and figured throwing myself in the deep end on Termux might learn me something. Hope yall have a nice day.

**Comment by codeandfire** [comment]
@cmaglie symlinking `/etc/resolv.conf` to `/run/systemd/resolve/stub-resolv.conf` worked for me ... Thank you very much for your help!

**Comment by cmaglie** [comment]
Thanks for the feedback!
Since it seems resolved in one way or another, I'm closing the issue.

**Comment by vitormaduro** [comment]
I was having the exact same issues when running IDE 2 in Fedora. Neither the Flatpak nor the AppImage versions worked, and manually running the `arduino-cli` command resulted in the same error.

My computer is connected to the internet, and there's no proxy between me and the outside world. Commands like `ping` worked correctly, but `dig` always timed out. I spent literally hours trying to find what was causing this issue (because my internet is working fine, I can even download the missing files manually from my browser).

**Eventually I gave up and started a Wi-Fi hotspot from my phone to share it's connection with my computer. Immediately the IDE managed to connect to the servers and downloaded everything, finishing the boot sequence in less than 10 seconds.**

Just wanted tell how I "fixed" my issue and managed to open the Arduino IDE 2 on Linux.

And just to help search engines index this comment, here are some keywords related to this issue[^1][^2][^3][^4][^5]:

[^1]: ERROR Failed to update platform, library indexes. Error: 13
[^2]: Arduino IDE 2 Linux hanging start
[^3]: Arduino IDE stuck logo
[^4]: code: 9, message: Error loading hardware
[^5]: Some indexes could not be updated.
